### DISCUSSION TITLE
一主两从的部署相较一主一从RT增加百倍

### DISCUSSION THREAD

**Discussion by orchidFang** [apache/rocketmq#6839] [General]
目前部署方式：
3namserver
3controller
3broker（1主2从）

broker.conf的配置 
flushDiskType=SYNC_FLUSH(同步刷盘）
allAckInSyncStateSet=true

利用mqadmin的clusterRT和CheckMsgSendRT来进行测试
在一主一从（2 broker）的情况下RT为2ms，
而在一主二从（3broker）的情况下RT高达450ms，
请问为什么在多了一个从节点的情况下RT增加这么多？

**Comment by RongtongJin** [comment]
> 目前部署方式： 3namserver 3controller 3broker（1主2从）
> 
> broker.conf的配置 flushDiskType=SYNC_FLUSH(同步刷盘） allAckInSyncStateSet=true
> 
> 利用mqadmin的clusterRT和CheckMsgSendRT来进行测试 在一主一从（2 broker）的情况下RT为2ms， 而在一主二从（3broker）的情况下RT高达450ms， 请问为什么在多了一个从节点的情况下RT增加这么多？

另一个从节点与其他两个节点之间是否隔的比较远？可以ping下。

**Reply by RongtongJin** [reply]
另外可以贴一下配置，三个节点的配置是否相同。理论上，延迟不会差这么多

**Reply by orchidFang** [reply]
1. 三个节点处于同一局域网内，ping为0.1ms左右。
2. 这三个broker节点，无论停掉哪个节点，只剩一个1主1从的情况下，其RT均为2ms左右，要是三个节点都启动，1主2从的方式RT则为400ms，所以判断不是某个节点的问题。
3. 三个节点的配置相同。
配置如下：
<img width="609" alt="image" src="https://github.com/apache/rocketmq/assets/6478487/6da67b54-a107-4ed5-936d-a841128c4f20">

**Comment by GenerousMan** [comment]
It is possible to encounter the situation you described when there are three replicas, which is caused by this piece of code:
![image](https://github.com/apache/rocketmq/assets/21073949/fe0a5f3f-82eb-43bb-a1e6-86577cacd15d)
 During the first acknowledgement check, the three replicas may not have all been confirmed yet, resulting in entering the next loop. When i=1, the return of the sending result will be delayed by one second.

**Comment by RongtongJin** [comment]
@orchidFang Fix in #6850
